spec/models/workshop_project_click_spec.rb

```diff
@@ -0,0 +1,45 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe 'V1::WorkshopProjectClick', type: :request do
+  describe 'GET /v1/projects/:id/redirect/:type_of' do
+    let(:project) { create(:workshop_project) }
+
+    # override in subtests
+    let(:headers) { {} }
+
+    before { get "/v1/projects/#{project.id}/redirect/live", headers: headers }
```

Comment:
Test both `live`, `code`, and invalid cases. Invalid cases should fail gracefully.